rc/detection/modeline.kak

```diff
@@ -86,13 +106,18 @@ define-command -hidden modeline-parse-impl %{
         # - the trailing text after the last option, and an optional ':' sign before it
         # It will also convert the ':' seperators beween the option=value pairs
         # More info: http://vimdoc.sourceforge.net/htmldoc/options.html#modeline
+        stop_looping=0
         printf %s "${kak_selection}" | sed          \
                 -e 's/^[^:]\{1,\}://'               \
                 -e 's/[ \t]*set\{0,1\}[ \t]//'      \
                 -e 's/:[^a-zA-Z0-9_=-]*$//'         \
                 -e 's/:/ /g'                        \
                 | tr ' ' '\n'                       \
-                | while read -r option; do
+                | while read -r option || stop_looping=1; do
+            if [ "${stop_looping}" = 1 ]; then
+                after_loop_ends
+                break
+            fi
```

Comment:
This confuses me (might be due to still being tired from NYE), why cannot the after_loop_ends logic be just written after the loop end (i.e. after the done on line 108 ?)

Reply:
The variables `last_vim_et`, `last_vim_sw` and `last_vim_ts` are modified during the loop, which is a subshell because of the pipe. If that code was written after the `done`, they would retain their initial values before the subshell of the `while` loop started.